mac_pw_pool/setup.sh

```diff
@@ -0,0 +1,112 @@
+#!/bin/bash
+
+# Setup and launch Cirrus-CI PW Pool node.  It must be called
+# with the env. var. `$POOLTOKEN` set.  It is assumed to be
+# running on a fresh AWS EC2 mac2.metal instance as `ec2-user`
+# or equivilent admin.  The instance must have both "metadata"
+# and "Allow tags in metadata" options enabled.
+
+set -eo pipefail
+
+GVPROXY_RELEASE_URL="https://github.com/containers/gvisor-tap-vsock/releases/latest/download/gvproxy-darwin"
+
+COMPLETION_FILE="$HOME/.setup.done"
+STARTED_FILE="$HOME/.setup.started"
+
+date -u -Iseconds >> "$STARTED_FILE"
+
+msg() { echo "##### ${1:-No message message provided}"; }
+die() { echo "ERROR: ${1:-No error message provided}"; exit 1; }
+
+[[ -n "$POOLTOKEN" ]] || \
+    die "Must be called with non-empty \$POOLTOKEN set."
+
+[[ ! -r "$COMPLETION_FILE" ]] || \
+    die "Appears setup script already ran at '$(cat $COMPLETION_FILE)'.  This script should not be called twice by automation."
+
+[[ "$USER" == "ec2-user" ]] || \
+    die "Expecting to execute as 'ec2-user'."
+
+msg "Configuring paths"
+grep -q homebrew /etc/paths || \
+    echo -e "/opt/homebrew/bin\n/opt/homebrew/opt/coreutils/libexec/gnubin\n$(cat /etc/paths)" \
+        | sudo tee /etc/paths > /dev/null
+
+# For whatever reason, when this script is run through ssh, the default
+# environment isn't loaded automatically.
+. /etc/profile
+
+msg "\$PATH=$PATH"
+
+msg "Installing podman-machine, testing, and CI deps. (~2m install time)"
+if [[ ! -x /usr/local/bin/gvproxy ]]; then
+    brew tap cfergeau/crc
+    brew install go go-md2man coreutils pstree vfkit cirruslabs/cli/cirrus
+
+    # Normally gvproxy is installed along with "podman" brew.  CI Tasks
+    # on this instance will be running from source builds, so gvproxy must
+    # be install from upstream release.
+    curl -sSLfO "$GVPROXY_RELEASE_URL"
+    sudo install -o root -g staff -m 0755 gvproxy-darwin /usr/local/bin/gvproxy
+    rm gvproxy-darwin
+fi
+
+msg "Adding/Configuring PW User"
+# Ref: https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/instancedata-data-retrieval.html
+PWINST=$(curl -sSLf http://instance-data/latest/meta-data/tags/instance/Name)
+PWUSER=$PWINST-worker
+[[ -n "$PWINST" ]] || \
+    die "Unexpectedly empty instance name, is metadata tag access enabled?"
+
+PWCFG=$(mktemp /private/tmp/${PWINST}_cfg_XXXXXXXX)
+PWLOG="/private/tmp/${PWUSER}.log"
+
+# CI effectively allows unmitigated access to run or host any
+# process or content on this instance as $PWUSER.  Limit the
+# potential blast-radius of any nefarious use by restricting
+# the lifetime of the instance.  If this ends up disturbing
+# a running task, Cirrus will automatically retry on another
+# available pool instance. Shutdown instance after this many
+# hours servicing the pool.  Note: It's randomized slightly
+# to prevent instances going down at similar times.
+_rndadj=$((RANDOM%8-4))  # +/- 4 hours
+PWLIFE=$((24+$_rndadj))
```

Comment:
Decided we probably don't want random/unpredictable things making life more complex, fixing this to be a simple 21-hour limit.